### ISSUE TITLE
Support convolutions with string-valued padding

### ISSUE THREAD

**Issue by f-dangel** [f-dangel/singd#35]
(follow-up from #34)

The current version of `_extract_patches` does not support string-valued padding, but only 2-tuples, although strings are allowed to specify the padding in `torch.nn.Conv2d`.

We could use [this](https://github.com/f-dangel/einconv/blob/master/einconv/utils.py#L97-L132) function from my `einconv` library to compute the left and right paddings given the convolution's hyper-parameters.
